### PR TITLE
Implement accept4

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

**Review by Geertiebear — CHANGES_REQUESTED** [review]
Please add a test.

**Comment by no92** [comment]
Rebased, added linux sysdeps and added a test.

File: options/posix/include/mlibc/posix-sysdeps.hpp [hunk 1]
```diff
@@ -149,7 +149,7 @@ int sys_vm_unmap(void *pointer, size_t size);
 [[gnu::weak]] int sys_sigaction(int, const struct sigaction *__restrict,
 		struct sigaction *__restrict);
 [[gnu::weak]] int sys_kill(int, int);
-[[gnu::weak]] int sys_accept(int fd, int *newfd);
+[[gnu::weak]] int sys_accept(int fd, int *newfd, struct sockaddr *addr_ptr, socklen_t *addr_length);
```
Comment (Geertiebear) [inline options/posix/include/mlibc/posix-sysdeps.hpp:152]: Could you please fix all sysdeps where ``sys_accept`` is implemented?

**Review by Geertiebear — CHANGES_REQUESTED** [review]

**Review by Geertiebear — APPROVED** [review]
LGTM.

**Pull request merged**